Week/index.ts

```diff
@@ -0,0 +1,54 @@
+import { DayOfWeek } from "DayOfWeek"
+import { isly } from "isly"
+import { Date } from "../Date"
+import { Number as WeekNumber } from "./Number"
+import { Parts as WeekParts } from "./Parts"
+
+export type Week = `${number}-W${number}`
+
+export namespace Week {
+	export import Number = WeekNumber
+	export import Parts = WeekParts
+	export const type = isly.named("Week", isly.string<Week>(/^\d{4}-W\d{2}$/))
```

Comment:
This accepts `"2024-W99"` as a valid week.

Reply:
If the function's below will handle `"2024-W00"` and `"2024-W99"` as the equivalent as `"2023-W52`" and `"2025-W47`" then I guess the definition is fine.

Reply:
I'm ambivalent on having the is function operating like a parse function. It is often to much compilation for the purpose of just checking between types. But I did anyhow narrow it down as we already have with other string based types in isoly.